app/src/main/java/io/appium/settings/location/LocationFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.appium.settings.location;
+
+import android.location.Location;
+import android.os.Build;
+import android.os.SystemClock;
+
+public class LocationFactory {
+
+    private double latitude;
+    private double longitude;
+    private double altitude;
+
+
+    public synchronized Location createLocation(String providerName, float accuracy) {
+        Location l = new Location(providerName);
+        l.setAccuracy(accuracy);
+
+        l.setLatitude(latitude);
```

Comment:
what happens if setLocation is not called before createLocation ?

Reply:
Do you mean if we create Location objects that never get successfully set as mock locations?

Reply:
I don't know, just asking